Cabal/Distribution/Backpack/ModSubst.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE PatternGuards #-}
+
+-- | A type class 'ModSubst' for objects which can have 'ModuleSubst'
+-- applied to them.
+
+module Distribution.Backpack.ModSubst (
```

Comment:
Would it be clearer if we rearranged, and put the ModSubst type and class here, and imported that from Package and defined the instances in Package. I feel we could clump things together more nicely. It feels odd that ModuleSubst and its related utils are not defined here.